.github/workflows/build.yml

```
@@ -0,0 +1,17 @@
+name: build-all-configs
+on: [push]
+jobs:
+  build-on-osx-for-osx:
+    runs-on: macos-10.15
+    steps:
+    - uses: actions/checkout@v2
+    - id: configure-cmake
```

You don't have to set an `id` for each job. `id`s are only required if you [want to reference a step from another one](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsid).

I'd prefer it if the steps [had a name](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idname) that describes whats going on, e.g.:
```suggestion
    - name: Configure build
```

---

a, that's why I got the error when I ised names with spaces :-) , will change